### PR TITLE
e2e_apps: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

e2e_apps: stop using deprecated framework.ExpectEqual

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by carlory** [comment]
Can you take a look at it?

/cc @pohly

File: test/e2e/apps/rc.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ var _ = SIGDescribe("ReplicationController", func() {
 			ginkgo.By("patching ReplicationController")
 			testRcPatched, err := f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).Patch(ctx, testRcName, types.StrategicMergePatchType, []byte(rcLabelPatchPayload), metav1.PatchOptions{})
 			framework.ExpectNoError(err, "Failed to patch ReplicationController")
-			framework.ExpectEqual(testRcPatched.ObjectMeta.Labels["test-rc"], "patched", "failed to patch RC")
+			gomega.Expect(testRcPatched.ObjectMeta.Labels["test-rc"]).To(gomega.Equal("patched"), "failed to patch RC")
```
Comment (pohly) [inline test/e2e/apps/rc.go:209]: `gomega.HaveKeyWithValue` would be useful here, too.

**Review by pohly — CHANGES_REQUESTED** [review]

**Review by pohly — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**